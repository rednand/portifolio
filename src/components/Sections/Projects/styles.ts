import styled from 'styled-components'

export const Title = styled.h1`
  font-size: 2.5rem;
  color: #fefefe;
  margin-bottom: 4%;

  @media (max-width: 600px) {
    margin: 8% 0;
  }
`
