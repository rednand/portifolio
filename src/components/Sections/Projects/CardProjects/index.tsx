import projects from '../../../../data/projects'
import * as Style from './styles'
import { FiGithub } from 'react-icons/fi'
import { useState } from 'react'

const CardProjects = () => {
  const [showIconGithub, setShowIconGithub] = useState<null | number>(null)

  return (
    <Style.ContainerCard data-testid="card-projects">
      {projects.map((project, index) => (
        <Style.Card
          key={index}
          onMouseEnter={() => setShowIconGithub(index)}
          onMouseLeave={() => setShowIconGithub(null)}
        >
          <Style.CardImage src={project.image} alt={project.projeto} />
          <Style.CardContent>
            <Style.CardTitle>
              {project.link ? (
                <a
                  href={project.link}
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  {project.projeto}
                </a>
              ) : (
                project.projeto
              )}
            </Style.CardTitle>
            <Style.CardDescription>{project.descricao}</Style.CardDescription>
          </Style.CardContent>
          {showIconGithub === index && (
            <Style.CardButtonContainer>
              <Style.CardButton
                href={project.github}
                target="_blank"
                rel="noopener noreferrer"
              >
                <FiGithub size={15} color="#48d8a4" />
              </Style.CardButton>
            </Style.CardButtonContainer>
          )}
        </Style.Card>
      ))}
    </Style.ContainerCard>
  )
}

export default CardProjects
