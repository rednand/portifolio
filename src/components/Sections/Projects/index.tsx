import * as S from '../../Wrapper'
import * as Style from './styles'
import CardProjects from './CardProjects'

const Projects = () => {
  return (
    <S.Wrapper>
      <Style.Title>Projetos</Style.Title>
      <CardProjects />
    </S.Wrapper>
  )
}

export default Projects
