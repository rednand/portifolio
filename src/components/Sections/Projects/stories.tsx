import { Meta, StoryObj } from '@storybook/react'
import Projects from '.'

export default {
  title: 'Projects',
  component: Projects
} as Meta

export const Default: StoryObj = {}
