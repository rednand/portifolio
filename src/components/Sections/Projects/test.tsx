import { render, screen } from '@testing-library/react'
import Projects from '.'

describe('<Projects />', () => {
  it('should render the heading', () => {
    const { container } = render(<Projects />)

    expect(
      screen.getByRole('heading', { name: /Projetos/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the CardProjects component', async () => {
    render(<Projects />)
    const cardProjects = await screen.findByTestId('card-projects')
    expect(cardProjects).toBeInTheDocument()
  })
})
