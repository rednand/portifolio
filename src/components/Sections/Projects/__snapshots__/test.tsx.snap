// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render the heading 1`] = `
.c0 {
  height: 100vh;
  padding: 10vh;
  background: #2c2a32;
  font-family: 'Segoe UI',Roboto,'Helvetica Neue',sans-serif;
  margin: auto;
  scroll-snap-align: start;
}

.c1 {
  font-size: 2.5rem;
  color: #fefefe;
  margin-bottom: 4%;
}

.c2 {
  display: flex;
  flex-wrap: wrap;
  gap: 20px;
  margin-bottom: 5%;
  justify-content: flex-start;
}

.c3 {
  padding: 20px;
  width: 300px;
  transition: transform 0.3s;
}

.c3:hover {
  transform: translateY(-5px);
  border-bottom: 5px solid #48d8a4;
}

.c4 {
  width: 100%;
  height: 150px;
  object-fit: fill;
  border-radius: 8px 8px 0 0;
  opacity: 0.5;
}

.c4:hover {
  opacity: 1;
}

.c5 {
  padding: 10px;
}

.c6 {
  font-size: 1.5rem;
  color: #fefefe;
  margin: 5% 0;
}

.c6 a {
  text-decoration: none;
  color: inherit;
  transition: color 0.3s;
}

.c6 a:hover {
  color: #48d8a4;
}

.c7 {
  font-size: 1rem;
  color: #cccccc;
  line-height: 1.5;
}

@media (max-width: 600px) {
  .c0 {
    padding: 3vh;
  }
}

@media (min-width: 600px) and (max-width: 1000px) {
  .c0 {
    height: 110vh;
  }
}

@media (max-width: 600px) {
  .c1 {
    margin: 8% 0;
  }
}

@media (max-width: 600px) {
  .c2 {
    gap: 10px;
    flex-wrap: nowrap;
    overflow-x: auto;
  }
}

@media (min-width: 600px) and (max-width: 1000px) {
  .c2 {
    overflow: auto;
    height: 90vh;
  }
}

@media (max-width: 600px) {
  .c3 {
    width: 300px;
    padding: 0;
    margin-bottom: 5%;
  }
}

@media (max-width: 414px) {
  .c4 {
    width: 80vw;
  }
}

@media (max-width: 600px) {
  .c4 {
    width: 50vw;
  }
}

<section
  class="c0"
>
  <h1
    class="c1"
  >
    Projetos
  </h1>
  <div
    class="c2"
    data-testid="card-projects"
  >
    <div
      class="c3"
    >
      <img
        alt="Se Liga, Hein"
        class="c4"
        src="/images/se-liga.png"
      />
      <div
        class="c5"
      >
        <h3
          class="c6"
        >
          <a
            href="https://se-liga-hein.vercel.app/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Se Liga, Hein
          </a>
        </h3>
        <p
          class="c7"
        >
          Projeto desenvolvido no curso de Desenvolvimento Web Front-End Node da Digital House.
        </p>
      </div>
    </div>
    <div
      class="c3"
    >
      <img
        alt="Aplicativo Web VHS 404"
        class="c4"
        src="/images/app-vhs404.png"
      />
      <div
        class="c5"
      >
        <h3
          class="c6"
        >
          <a
            href="https://vhs404.vercel.app/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Aplicativo Web VHS 404
          </a>
        </h3>
        <p
          class="c7"
        >
          Aplicativo de lançamentos de filmes de terror desenvolvido durante estágio como desenvolvedor Front-End.
        </p>
      </div>
    </div>
    <div
      class="c3"
    >
      <img
        alt="API do Aplicativo Web VHS 404"
        class="c4"
        src="/images/api-vhs404.png"
      />
      <div
        class="c5"
      >
        <h3
          class="c6"
        >
          <a
            href="https://api-vhs404.vercel.app/"
            rel="noopener noreferrer"
            target="_blank"
          >
            API do Aplicativo Web VHS 404
          </a>
        </h3>
        <p
          class="c7"
        >
          API desenvolvida para o aplicativo web VHS 404, criada durante estágio como desenvolvedor Front-End.
        </p>
      </div>
    </div>
    <div
      class="c3"
    >
      <img
        alt="Desenvolvedor Front-End"
        class="c4"
        src="/images/corebiz.png"
      />
      <div
        class="c5"
      >
        <h3
          class="c6"
        >
          Desenvolvedor Front-End
        </h3>
        <p
          class="c7"
        >
          Landing page criada durante o bootcamp Hiring Coders, que faz a integração da API do VTEX IO com a API da AWS.
        </p>
      </div>
    </div>
  </div>
</section>
`;
